internal/configs/version2/nginx-plus.virtualserver.tmpl

```diff
@@ -429,6 +429,12 @@ server {
         {{ $proxyOrGRPC }}_set_header username $jwt_claim_sub;
         {{ end }}
 
+        {{ with $oidc := $s.OIDC }}
+            {{ if $oidc.AccessTokenEnable }}
+        {{ $proxyOrGRPC }}_set_header Authorization "Bearer $access_token";
+            {{ end }}
+        {{ end }}
```

Comment:
```suggestion
        {{- if $s.OIDC.AccessTokenEnable }}
        {{ $proxyOrGRPC }}_set_header Authorization "Bearer $access_token";
        {{- end }}
        {{ end }}
```

@shawnhankim after more testing, we need to make sure that OIDC is enabled at all before writing the directive in the location block. The `$l.OIDC` is true only if there's a policy configured, so it's safe to move it inside the block. Sorry for the back and forth, but after this change and a rebase with `main`, we're good to merge 🚀

Reply:
Thanks @lucacome for your careful review and time. I am happy to change it. 😸 I have changed, rebased and manually tested it.